test/unit/lodash.spec.js

```diff
@@ -0,0 +1,275 @@
+        });
+        it('should return first matched element for name property', function () {
+            var params = datasets.find2();
+            var array = params[0];
+            expect(array.length).to.eql(3);
+            array.forEach(function (n) {
+                expect(_.has(n, 'NAME')).to.eql(true);
+                expect(_.has(n, 'active')).to.eql(true);
+            });
+            expect(_.find.apply(_, datasets.find2())).to.eql({
+                NAME: 'Rudi',
+                age: 15,
+                active: true,
+                gender: 'M'
+            });
+        });
+    });
+
+    describe('groupBy', function () {
+        it('should group floating-point numbers with Math function', function () {
+            var params = datasets.groupBy();
+            var array = params[0];
+            expect(array.length).to.eql(7);
+            expect(_.groupBy.apply(_, datasets.groupBy())).to.eql({
+                3: [3.3],
+                4: [4.2, 4.6],
+                6: [6.1, 6.3],
+                8: [8.3, 8.9]
+            });
+        });
+        it('should group all given things by length', function () {
+            var params = datasets.groupBy2();
+            var array = params[0];
+            expect(array.length).to.eql(6);
+            expect(_.groupBy.apply(_, datasets.groupBy2())).to.eql({
+                3: ['xyz', 'xyz'],
+                5: ['combi'],
+                undefined: [2, 4, 5]
+            });
+        });
+    });
+
+    describe('includes', function () {
+        it('should include first value in the object', function () {
+            var params = datasets.includes();
+            var array = params[0];
+            var predicate = params[1];
+            expect(Object.keys(array).length).to.eql(4);
+            expect(_.has(array, 'firstName')).to.eql(true);
+            expect(_.has(array, 'lastName')).to.eql(true);
+            expect(_.has(array, 'users')).to.eql(true);
+            expect(_.has(array, 'running')).to.eql(true);
+            expect(_.includes.apply(_, [array, predicate[0]])).to.eql(true);
+            expect(_.includes.apply(_, [array, predicate[1]])).to.eql(false);
+        });
+        it('should include \'stabli\' word in array of random words', function () {
+            var params = datasets.includes2();
+            var array = params[0];
+            var predicate = params[1];
+            expect(array.length).to.eql(4);
+            expect(array).to.contain('antidisestablishmentarianism');
+            expect(array).to.not.contain('stabli');
+            array.forEach(function (element, index) {
+                if (index !== 3) {
+                    expect(_.includes.apply(_, [element, predicate])).to.eql(true);
+                } else {
+                    return false;
+                }
+            });
+            expect(_.includes.apply(_, [array[3], predicate])).to.eql(false);
+        });
+    });
+
+    describe('map', function () {
```

Comment:
map: For the first case I was again able to pass test without having done exercise. It is great that you've checked if second parameter is a string, but have you considered it may be an empty or whaterver string? Moreover, why do you need two datasets? It only complicated code but tests nothing about lodash knowledge to be honest. What you actually do is to compare result with result and so on. It has NO MEANING, you could as well do expect(true).to.eql(true). Second example is even worse, you only return array that is provided as parameter and tests are passed. Secondly, we are supposed to indicate what guarded methods are and show what can mapping be used for. Lastly, we need to test multiple parameter types.